gradio/utils.py

```diff
@@ -1280,3 +1280,25 @@ async def function_wrapper(*args, **kwargs):
         return f(*args, **kwargs)
 
     return function_wrapper
+
+
+class FileSize:
+    B = 1
+    KB = 1024 * B
+    MB = 1024 * KB
+    GB = 1024 * MB
+    TB = 1024 * GB
+
+
+def _parse_file_size(size: str | int | None) -> int | None:
+    if isinstance(size, int) or size is None:
+        return size
+
+    last_digit_index = next(
+        (i for i, c in enumerate(size) if not c.isdigit()), len(size)
+    )
+    size_int, unit = int(size[:last_digit_index]), size[last_digit_index:].upper()
```

Comment:
Perhaps we should strip these to be a bit more robust in case users do something like "5 mb"
```suggestion
    size_int, unit = int(size[:last_digit_index].strip()), size[last_digit_index:].strip().upper()
```

Reply:
Added a test for this case too